lib/testutils/resources.go

```diff
@@ -199,8 +199,29 @@ func (t *testResourceWatcher) expectEvents(kind string, anyOrder bool, expectedE
 	// events, so protect against that scenario - we'll check later once we've
 	// constructed useful diagnostics.
 	var actualEvents []watch.Event
+	log.Infof("Received %s events, expected %d", len(t.events), len(expectedEvents))
+	// CASEY: TODO - Can we assert this generally?
+	Expect(len(t.events)).To(Equal(len(expectedEvents)), "Wrong number of events received")
```

Comment:
Tests seem to pass with this assertion, so might be worth adding it in. WDYT?

Reply:
If we are keeping this, we should do it in a way that means we'll still benefit from the other added diags below.  (As it stands, I believe the added code below will never be executed.)